core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2023,6 +2023,22 @@ package object config {
       .stringConf
       .createWithDefaultString("PWR")
 
+  private[spark] val EXECUTOR_DECOMMISSION_BATCH_INTERVAL =
+    ConfigBuilder("spark.executor.decommission.batchInterval")
+      .doc("Executors are decommissioned in batched to avoid overloading network bandwidth in" +
+        " migrating rdd and shuffle data. This config sets the interval between batches.")
+      .version("3.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefault(30000)
```

Comment:
Azure gives a very limited time of just 30s for spot instance removal. 
With these defaults only 3 nodes will get a chance to evacuate shuffle/cache data. 
(9 nodes on AWS with the 90s notification)
Any chance to make this (much) smaller?

Reply:
Thanks for reviewing this. I have set this interval to much smaller value as 3s. 

I am not sure if we can set this to a value which works in all scenarios. It should depend on the nature of the application and ultimately the user will have freedom to set this.